src/credentials/sso/ssoAccessTokenProvider.ts

```diff
@@ -122,7 +123,12 @@ export class SsoAccessTokenProvider {
 
             return refreshed
         } catch (err) {
-            telemetry.aws_refreshCredentials.emit({
+            const span = telemetry.aws_refreshCredentials
+            if (DevSettings.instance.get('reportRequestIds', false)) {
+                span.record({ requestId: getRequestId(err) } as any)
+            }
```

Comment:
Added gated logic to record `requestId` until we get confirmation on this.

Reply:
could gate this simply on `isDevMode` https://github.com/aws/aws-toolkit-vscode/blob/79fc42cf92fecebfec9a8c4dbf425bedb2cb3d56/src/dev/activation.ts#L91